hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3007,25 +3007,33 @@ public String getRpcPort() {
   }
 
   @Override
-  public String getRatisRoles() {
+  public List<Map<String, String>> getRatisRoles() {
     List<ServiceInfo> serviceList;
+    List<Map<String, String>> resultList = new ArrayList<>();
+    Map<String, String> messageException = new HashMap<>();
     int port = omNodeDetails.getRatisPort();
     RaftPeer leaderId;
     if (isRatisEnabled) {
       try {
         leaderId = omRatisServer.getLeader();
         if (leaderId == null) {
           LOG.error("No leader found");
-          return "Exception: Not a leader";
+          messageException.put("Message", "Exception: Not a leader");
+          resultList.add(messageException);
+          return resultList;
```

Comment:
This solution could work, but currently if we have an exception, we will return with this list of 1 map, which only has one element. But in the `om-overview.html` file we are expecting the 4 attributes of the OM role and we are explicitly requesting the 2nd and 3rd element's values from the map, so this won't work this way. Maybe if we only have one element in the map (or if we switch it to list, one element there) we should handle that on the UI and show only the message without the table.